compiler/src/iree/compiler/Codegen/Common/OptimizeVectorTransferPass.cpp

```diff
@@ -91,7 +91,8 @@ struct OptimizeVectorTransferPass final
       // Workaround, run loop invariant code motion before hoist redundant
       // vector transfer to workaround a bug upstream.
       loopInvariantCodeMotion(funcOp);
-      linalg::hoistRedundantVectorTransfers(cast<func::FuncOp>(funcOp));
+      linalg::hoistRedundantVectorTransfers(cast<func::FuncOp>(funcOp),
+                                            /*verifyNonZeroTrip=*/true);
```

Comment:
We can make this false for the integrate which keeps the same behavior in the integrate commit. Then you can send another patch to enable this.

Reply:
Oh I see, there are other changes already break the tests and the change here fixes the issue.